integrations/beacon-blobs/src/runtime.ts

```diff
@@ -0,0 +1,175 @@
+import { DataItem, IRuntime, Validator, VOTE } from "@kyvejs/protocol";
+import { name, version } from "../package.json";
+import axios from "axios";
+import { createVersionedHash, getTransactionByHash } from "../utils/utils";
+import { providers } from "ethers";
+import { hexValue } from "ethers/lib/utils";
+
+
+// Beacon Blobs config
+interface IConfig {
+  consensusRPC: string;
+  executionRPC: string;
+  finality: number;
+  genesisTime: number;
+  sequencer: string[];
+}
+
+export default class BeaconBlobs implements IRuntime {
+  public name = name;
+  public version = version;
+  public config!: IConfig;
+
+  async validateSetConfig(rawConfig: string): Promise<void> {
+    const config: IConfig = JSON.parse(rawConfig);
+
+    if (!config.finality) {
+      throw new Error(`Config does not have finality defined`);
+    }
+
+    if (!config.consensusRPC) {
+      throw new Error(`Config does not have property "consensusRPC" defined`);
+    }
+
+    if (!config.executionRPC) {
+      throw new Error(`Config does not have property "executionRPC" defined`);
+    }
+
+    if (!config.genesisTime) {
+      throw new Error(`Config does not have property "genesisTime" defined`);
+    }
+
+    if (!config.sequencer.length) {
+      throw new Error(`Config does not have property "sequencer" defined`);
+    }
+
+    if (process.env.KYVEJS_BEACON_BLOBS_EXECUTION_RPC) {
+      config.executionRPC = process.env.KYVEJS_BEACON_BLOBS_EXECUTION_RPC;
+
+      console.log("set config executionRPC to", config.executionRPC)
+    }
+
+    if (process.env.KYVEJS_BEACON_BLOBS_CONSENSUS_RPC) {
+      config.consensusRPC = process.env.KYVEJS_BEACON_BLOBS_CONSENSUS_RPC;
+
+      console.log("set config consensusRPC to", config.consensusRPC)
+    }
+
+
+    this.config = config;
+  }
+
+  async  getDataItem(_: Validator, key: string): Promise<any> {
+    const provider = new providers.StaticJsonRpcProvider(this.config.executionRPC);
+
+    const currentHeight = await provider.getBlockNumber();
+
+    const hexKey = hexValue(+key);
+
+    const block = await provider.getBlockWithTransactions(hexKey);
+
+    // only validate if current height is already 'finalized'
+    if (block.number >= currentHeight - 256) {
+      throw new Error(
+        `Finality not reached yet; waiting for next block`
+      )
+    }
+
+    // Get all type3 transactions that has been sent to the sequencer inbox
+    const filteredTransactions = block.transactions.filter(
+      (tx) => {
+        if (tx.to) {
```

Comment:
could be shorter with `return tx.type == 3 && tx.to && this.config.sequencer.includes(tx.to)`